span.go

```diff
@@ -392,6 +432,7 @@ type SpanData struct {
 	stackTraceLimit        int
 	timestamp              time.Time
 	childrenTimer          childrenTimer
+	exit                   bool
```

Comment:
SpanData is removed when the span is ended, so I think we'll need to move `exit` to the Span type as a sibling to the Span.TraceContext field. That way we can still tell if a span is/was an exit span even after it's ended.